uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js

```diff
@@ -24,8 +25,33 @@ module.exports = (function () {
       }.bind(this));
     },
 
+    /**
+     * Opens the Leave Request Modal for a new request of the given type
+     *
+     * @param  {String} requestType leave|sickness|toil
+     * @return {Promise}
+     */
+    newRequest: function (requestType) {
+      var casper = this.casper;
+      var requestTypes = ['leave', 'sickness', 'toil']; // must be in the same quantity and order as in UI
+      var requestTypeButtonIndex = requestTypes.indexOf(requestType) + 1;
+      var actionDropdownSelector = 'leave-request-record-actions';
+      var actionButtonSelector = actionDropdownSelector + ' .dropdown-menu a:nth-child(' + requestTypeButtonIndex + ')';
```

Comment:
If you use jQuery, you can use the `.contains` method. For example you could do something like this:

```js
casper.evaluate(function (actionDropdownSelector, requestType) {
  var requestTypeButton = jQuery(actionDropdownSelector + ' .dropdown-menu a:contains(' + requestType + ')');

  requestTypeButton.click(); 
}, actionDropdownSelector, requestType);
```

Reply:
The whole point was not to use jQuery.